sdks/wasm/src/library_mono.js

```diff
@@ -441,31 +484,53 @@ var MonoSupportLib = {
 		},
 	},
 
-	mono_wasm_add_bool_var: function(var_value) {
-		MONO.var_info.push({
-			value: {
-				type: "boolean",
-				value: var_value != 0,
-			}
-		});
-	},
+	mono_wasm_add_typed_value: function (type, str_value, int_value) {
+		var type_str = Module.UTF8ToString (type);
 
-	mono_wasm_add_char_var: function(var_value) {
-		MONO.var_info.push({
-			value: {
-				type: "symbol",
-				value: `${var_value} '${String.fromCharCode (var_value)}'`
-			}
-		});
-	},
+		switch (type_str) {
+		case "bool":
+			MONO.var_info.push ({
+				value: {
+					type: "boolean",
+					value: int_value != 0
+				}
+			});
+			break;
 
-	mono_wasm_add_number_var: function(var_value) {
-		MONO.var_info.push({
-			value: {
-				type: "number",
-				value: var_value,
-			}
-		});
+		case "char":
+			MONO.var_info.push ({
+				value: {
+					type: "symbol",
+					value: `${int_value} '${String.fromCharCode (int_value)}'`
+				}
+			});
+			break;
+
+		case "number":
+			MONO.var_info.push ({
+				value: {
+					type: "number",
+					value: int_value
+				}
+			});
+			break;
+
+		case "string":
+			MONO._mono_wasm_add_string_var (str_value);
+			break;
+
+		case "getter":
+			MONO._mono_wasm_add_getter_var (str_value);
+			break;
+
+		case "array":
+			MONO._mono_wasm_add_array_var (str_value, int_value);
+			break;
+
+		default:
+			throw new Error (`Unknown type for value: ${type}`);
```

Comment:
an error here will probably surface in a really bad way maybe just log it and set a string to the value?